app/services/create_account.rb

```diff
@@ -1,56 +1,63 @@
-class CreateAccount < ApplicationService
-  def initialize(payload, from_fintera = false)
-    @payload = payload
-    @from_fintera = from_fintera
-    @errors = []
-  end
-
-  def call
-    if !is_account_valid?
-      @errors << "Account is not valid"
-      Result.new(false, nil, @errors.join(","))
-    else
-      account = Account.new(account_params)
-      if account.save && User.insert_all(users_params(account))
-        Result.new(true, account)
-      else
-        @errors << account.errors.full_messages
-        Result.new(false, nil, @errors.join(","))
-      end
-    end
-  end
-
-  def is_account_valid?
-    return false if @payload.blank?
-
-    true
-  end
-
-  def account_params
-    if @from_fintera
-      {
-        name: @payload[:name],
-        active: true,
-      }
-    else
-      {
-        name: @payload[:name],
-        active: false,
-      }
-    end
-  end
-
-  def users_params(account)
-    @payload[:users].map do |user|
-      {
-        first_name: user[:first_name],
-        last_name: user[:last_name],
-        email: user[:email],
-        phone: user[:phone].to_s.gsub(/\D/, ""),
-        account_id: account.id,
-        created_at: Time.zone.now,
-        updated_at: Time.zone.now,
-      }
-    end
-  end
-end
+class CreateAccount < ApplicationService
+  def initialize(payload, from_fintera = false)
+    @payload = payload
+    @from_fintera = from_fintera
+    @errors = []
+    
+  end
+
+  def call
+    @errors << "Account is not valid" if @payload.blank?
+    return create_result if @errors.present?
+
+    @account = Account.create(account_params)
+    @errors << @account.errors.full_messages unless @account.save
+    return create_result if @errors.present?
+
+    entity = Entity.create(name: @payload[:entity_name], account: @account)
+    @errors << entity.errors.full_messages unless entity.save
+    return create_result if @errors.present?
+    
+    users = []
+    users_params.each do |user_param|
+      puts user_param[:email]
+      user = User.create(user_param)
+      @errors << user.errors.full_messages unless user.save
+      return create_result if @errors.present?
+
+      users << user
+    end
+
+    users.each do |user|
+      entity_user = EntityUser.create(user: user, entity: entity)
+      @errors << entity_user.errors.full_messages unless entity_user.save
+      return create_result if @errors.present?
+    end
+    return create_result
+  end
+
+  private
+
+  def create_result
+    return Result.new(true, @account) unless @errors.present?
+    return Result.new(false, nil, @errors.join(", "))
+  end
+
+  def account_params
+    return {name: @payload[:name], active: false} unless @from_fintera
+    {name: @payload[:name], active: true} 

```

Comment:
```suggestion
    return {name: @payload[:name], active: false} unless @from_fintera

    {name: @payload[:name], active: true} 
```
Aqui só um detalhe, costumamos deixar uma linha em branco após uma [GuardClause](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause).

Este é um dos pontos verificados pelo `rubocop`